src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -263,3 +264,215 @@ void Terminal::ColorSelection(const COORD, const COORD, const TextAttribute)
 {
     THROW_HR(E_NOTIMPL);
 }
+
+// Method Description:
+// - update _selection's endpoint according to the direction provided.
+//    It is moved according the the expansion mode.
+// Arguments:
+// - dir: the direction that the selection endpoint will attempt to move to
+// - mode: the selection expansion mode that the selection anchor will adhere to
+// Return Value:
+// - <none>
+void Terminal::MoveSelectionAnchor(Direction dir, SelectionExpansionMode mode, SelectionEndpoint target)
+{
+    COORD endpoint = ((target == SelectionEndpoint::Start) ? _selection->start : _selection->end);
+
+    switch (mode)
+    {
+    case SelectionExpansionMode::Cell:
+        _UpdateAnchorByCell(dir, endpoint);
+        break;
+    case SelectionExpansionMode::Word:
+        _UpdateAnchorByWord(dir, endpoint, target);
+        break;
+    case SelectionExpansionMode::Viewport:
+        _UpdateAnchorByViewport(dir, endpoint);
+        break;
+    case SelectionExpansionMode::Buffer:
+        _UpdateAnchorByBuffer(dir, endpoint);
+        break;
+    default:
+        // the provided selection expansion mode is not supported
+        THROW_HR(E_NOTIMPL);
+        return;
+    }
+
+    // scroll if necessary
+    bool notifyScroll = false;
+    if (endpoint.Y < _VisibleStartIndex())
+    {
+        _scrollOffset = ViewStartIndex() - endpoint.Y;
+        notifyScroll = true;
+    }
+    else if (endpoint.Y > _VisibleEndIndex())
+    {
+        _scrollOffset = std::max(0, ViewStartIndex() - endpoint.Y);
+        notifyScroll = true;
+    }
+
+    if (notifyScroll)
+    {
+        // TODO CARLOS: I need to learn how to do TriggerScroll here properly
+        _buffer->GetRenderTarget().TriggerRedrawAll();
+        _NotifyScrollEvent();
+    }
+
+    // update internal state of selection anchor and vertical offset
+    if (target == SelectionEndpoint::Start)
+    {
+        _selection->start = endpoint;
+    }
+    else
+    {
+        _selection->end = endpoint;
+    }
+}
+
+// Method Description:
+// - update the endSelectionPosition anchor by one cell according to the direction provided.
+// Arguments:
+// - dir: the direction that the selection anchor will attempt to move to
+// Return Value:
+// - <none>
+void Terminal::_UpdateAnchorByCell(Direction dir, COORD& anchor)

```

Comment:
Should these guys be taking `til::point`s now?

Reply:
Yeah. But I really should do it for all of the selection code. I'd rather do it all in one sweep. Created follow-up task #5814